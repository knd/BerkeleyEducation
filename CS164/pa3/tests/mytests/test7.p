contains(A,[A|R]).
contains(A,[B|R]) :- contains(A,R).
remove(A,[A|R],R).
remove(A,[F|R],[F|S]) :- remove(A,R,S).
append([A|B],Z,[A|W]) :- append(B,Z,W).
append([],A,A).
union([A|B],Z,W) :- contains(A,Z),  union(B,Z,W).
union([A|B],Z,[A|W]) :- union(B,Z,W), ~contains(A,Z).
union([],Z,Z).
union([a,b,c,d], [g,f,e,d], A)?
