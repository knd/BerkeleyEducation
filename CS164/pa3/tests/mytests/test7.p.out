Asking for solution 1
A = [a|[b|[c|[g|[f|[e|[d|[]]]]]]]]
Asking for solution 2
None
